lib/compat/wordpress-6.8/class-gutenberg-rest-post-archive-search-handler.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * REST API: WP_REST_Post_Archive_Search_Handler class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 6.8.0
+ */
+
+/**
+ * Core class representing a search handler for Post Archives in the REST API.
+ *
+ * @since 6.8.0
+ *
+ * @see WP_REST_Search_Handler
+ */
+class Gutenberg_REST_Post_Archive_Search_Handler extends WP_REST_Search_Handler {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.8.0
+	 */
+	public function __construct() {
+		$this->type = 'post-type-archive';
+	}
+
+	/**
+	 * Searches post-type archives for a given search request.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array {
+	 *     Associative array containing found IDs and total count for the matching search results.
+	 *
+	 *     @type int[]               $ids   Found post archive IDs.
+	 *     @type string|int|WP_Error $total Numeric string containing the number of post-type archives found, or WP_Error object.
```

Comment:
```suggestion
	 *     @type int $total The number of post-type archives found.
```

To my knowledge, `count()` [returns](https://www.php.net/manual/en/function.count.php) an integer.